analyzers/tests/SonarAnalyzer.UnitTest/Rules/SymbolicExecution/SymbolicExecutionRunnerTest.cs

```diff
@@ -407,24 +408,45 @@ private static void VerifyClassMainVB(string members)
 $@"Public Class Sample
     {members}
 End Class";
-            VerifyCode<VB.SymbolicExecutionRunner>(code, ProjectType.Product, ImmutableArray<ParseOptions>.Empty, null, MainScopeAssignmentRuleCheck.SMain, BinaryRuleCheck.SBinary);
+            VerifyCode<TestSERunnerVB>(code, ProjectType.Product, ImmutableArray<ParseOptions>.Empty, null, MainScopeAssignmentRuleCheck.SMain, BinaryRuleCheck.SBinary);
         }
 
         private static void VerifyCode<TRunner>(string code, ProjectType projectType, ImmutableArray<ParseOptions> parseOptions, string sonarProjectConfigPath, params DiagnosticDescriptor[] onlyRules)
-            where TRunner : SymbolicExecutionRunnerBase, new()
-        {
-            var sut = new TRunner();
-            sut.RegisterRule<BinaryRuleCheck>(BinaryRuleCheck.SBinary);
-            sut.RegisterRule<AllScopeAssignmentRuleCheck>(AllScopeAssignmentRuleCheck.SAll);
-            sut.RegisterRule<MainScopeAssignmentRuleCheck>(MainScopeAssignmentRuleCheck.SMain);
-            sut.RegisterRule<TestScopeAssignmentRuleCheck>(TestScopeAssignmentRuleCheck.STest);
-            new VerifierBuilder().AddAnalyzer(() => sut)
+            where TRunner : SymbolicExecutionRunnerBase, new() =>
+            new VerifierBuilder<TRunner>()
                 .AddReferences(TestHelper.ProjectTypeReference(projectType))
                 .AddSnippet(code)
                 .WithSonarProjectConfigPath(sonarProjectConfigPath)
                 .WithOptions(parseOptions)
                 .WithOnlyDiagnostics(onlyRules)
                 .Verify();
+
+        private class TestSERunnerCS : CS.SymbolicExecutionRunner
+        {
+            protected override ImmutableDictionary<DiagnosticDescriptor, RuleFactory> AllRules => ImmutableDictionary<DiagnosticDescriptor, RuleFactory>.Empty
+                .Add(BinaryRuleCheck.SBinary, CreateFactory<BinaryRuleCheck>())
+                .Add(AllScopeAssignmentRuleCheck.SAll, CreateFactory<AllScopeAssignmentRuleCheck>())
+                .Add(MainScopeAssignmentRuleCheck.SMain, CreateFactory<MainScopeAssignmentRuleCheck>())
+                .Add(TestScopeAssignmentRuleCheck.STest, CreateFactory<TestScopeAssignmentRuleCheck>());
+        }
+
+        private class TestSERunnerVB : VB.SymbolicExecutionRunner
+        {
+            protected override ImmutableDictionary<DiagnosticDescriptor, RuleFactory> AllRules => ImmutableDictionary<DiagnosticDescriptor, RuleFactory>.Empty
+                .Add(BinaryRuleCheck.SBinary, CreateFactory<BinaryRuleCheck>())
+                .Add(AllScopeAssignmentRuleCheck.SAll, CreateFactory<AllScopeAssignmentRuleCheck>())
+                .Add(MainScopeAssignmentRuleCheck.SMain, CreateFactory<MainScopeAssignmentRuleCheck>())
+                .Add(TestScopeAssignmentRuleCheck.STest, CreateFactory<TestScopeAssignmentRuleCheck>());
+        }
+
+        private class CustomSERunnerCS : CS.SymbolicExecutionRunner
```

Comment:
`Custom` does not provide more information about what is the behavior of this runner. What do you think about using `Configurable` instead? (we configure the runner by registering different rules).